java/src/main/java/com/google/appengine/tools/pipeline/impl/backend/CloudTasksTaskQueue.java

```diff
@@ -115,14 +107,22 @@ private TaskSpec ensureDelay(TaskSpec taskSpec, Duration delay) {
     }
   }
 
-  Collection<TaskReference> enqueue(@NonNull String queueName, Collection<TaskSpec> taskSpecs) {
+  @Override
+  public Collection<TaskReference> enqueue(@NonNull String queueName, Collection<TaskSpec> taskSpecs) {
     String queueLocation = cloudTasksLocationFromAppEngineLocation(appEngineServicesService.getLocation());
     QueueName queue = QueueName.of(appEngineEnvironment.getProjectId(), queueLocation, queueName);
+    Collection<TaskReference> taskReferences = new ArrayList<>();
     try (CloudTasksClient cloudTasksClient = cloudTasksClientProvider.get()) {
-      return taskSpecs.parallelStream() //q: this safe? efficient?
-        .map(taskSpec -> createIgnoringExisting(cloudTasksClient, queue, taskSpec))
-        .map(task -> TaskReference.of(queueName, TaskName.parse(task.getName()).getTask()))
-        .collect(Collectors.toList());
+      for (TaskSpec taskSpec : taskSpecs) {
```

Comment:
removed the parallelism, probably not very efficient as not many tasks being enqueued at the same time right? 
this worked as an all-or-nothing, but being in parallel, if a task was created there was no way to know it in case of an error. So doing it sequentially and added side effect of deleting tasks if something went wrong.

Reply:
:warning: my main concern here is how this is going to behave on retries, as tasks should be tombstoned for 7 days... @eschultink I think the code does generate random task names for the incremental tasks, that should be the main use case here, so probably OK but would like your thoughts.

Reply:
well, ~40 tasks enqueued in MR case and I'm not sure what the benefit of doing API calls in series is.

Reply:
ok, can add it back, but what we really need is to capture all the task that have been created.
if enqueuing 10, 5 go through and 1 fail, as things stand now, this triggers an exception and we never got the references for the first 5 to later do the clean-up.